### PR TITLE
Expand tilde in `read_edf()`

### LINKED ISSUE #22: Consider expanding `~` in a file path
If the path provided to `edfio.read_edf()` contains a `~`, I get a `FileNotFoundError`. The reason is that the tilde is not expanded to the user's home directory. I think it would be useful if the function expanded paths internally.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #22.

### CODE REVIEW

File: tests/test_edf.py [hunk 1]
```diff
@@ -619,6 +619,11 @@ def test_read_edf_str():
     assert read_edf(str(EDF_FILE)).num_signals == 7
 
 
+def test_read_edf_expanduser():
+    with pytest.raises(FileNotFoundError, match="^[^~]*$"):
```
Comment (hofaflo) [inline tests/test_edf.py:623]: Nice solution for testing this!

File: tests/test_edf.py [hunk 2]
```diff
@@ -619,6 +619,11 @@ def test_read_edf_str():
...
+        read_edf("~/test.edf")
```
Comment (hofaflo) [inline tests/test_edf.py:624]: Maybe we should use a filename which is extremely unlikely to actually exist  :thinking:

**Review by hofaflo — COMMENTED** [review]
Thanks for the PR! Could you add a test with a symlink in a tempdir to cover `.resolve()` and a changelog entry?

File: tests/test_edf.py (hunk 2 above)
Comment (hofaflo) [reply tests/test_edf.py:624]: Yes exactly, I'm thinking about local execution. I'd use something like `edfio_testfile_that_should_not_exist.edf`. I agree, but the only other option I see is to use (os-dependent) environment variables to change what `~` expands to, so I'd rather go with this :D

Comment (cbrnr) [reply tests/test_edf.py:624]: Well, a third option would be to not test this. I can see your point to make sure no one reverts this expansion in the future, but then I'd just fix it if it comes up.

**Comment by hofaflo** [comment]
I just tried to write a test using a symlinked file or directory (to make sure `.resolve()` works), but apparently `.resolve()` is not necessary, as it works either way (at least on linux) :shrug:

Another thought: we should probably use `.expanduser()` in `Edf.write()` as well

**Comment by cbrnr** [comment]
> I just tried to write a test using a symlinked file or directory (to make sure `.resolve()` works), but apparently `.resolve()` is not necessary, as it works either way (at least on linux) 🤷

No idea what this actually does, it probably doesn't hurt to leave it in. But it's up to you, let me know if you want me to remove it.

I'll add it to `Edf.write()`, sure.

**Review by cbrnr — COMMENTED** [review]
BTW, I'm running `ruff --preview`, and I get [PLR0904](https://docs.astral.sh/ruff/rules/too-many-public-methods/) for the `Edf` class. Just wanted to give you a heads up before this rule becomes stable.

**Comment by cbrnr** [comment]
(I didn't mean to put my last comment in a review. I meant to write a normal comment, but the UI from within VS Code is... not ideal.)

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## [Unreleased]
 
+### Added
+- Expand `~` (tilde) to the user's home directory in `edf_file` argument of `read_edf()` ([#23](https://github.com/the-siesta-group/edfio/pull/23)).
```
Comment (hofaflo) [inline CHANGELOG.md:6]:
```suggestion
- Expand `~` (tilde) to the user's home directory in `edf_file` argument of `read_edf()` and `target` argument of `Edf.write()` ([#23](https://github.com/the-siesta-group/edfio/pull/23)).
```

**Review by hofaflo — APPROVED** [review]

**Comment by hofaflo** [comment]
Thanks @cbrnr!

**Pull request merged**